contracts/emissions/EmissionsController.sol

```diff
@@ -203,19 +203,19 @@ contract EmissionsController is IGovernanceHook, Initializable, ImmutableModule
      * @return emissionForEpoch Units of MTA to be distributed at this epoch
      */
     function topLineEmission(uint32 epoch) public view returns (uint256 emissionForEpoch) {
-        // e.g. week 1, A = -166000e12, B = 180000e12, C = -180000e12, D = 166000e12
-        // e.g. epochDelta = 1e18
-        uint128 epochDelta = (epoch - epochs.startEpoch) * 1e18;
-        // e.g. x = 1e18 / 312e6 = 3205128205
-        int256 x = SafeCast.toInt256(epochDelta / EPOCHS);
+        require(epochs.startEpoch < epoch && epoch<= epochs.startEpoch + 312 , "Wrong epoch number");
```

Comment:
Not sure about the upper limit, epoch<= epochs.startEpoch + 312 , it avoids to return negative values with the current configuration,  but this could change if the configuration changes

Reply:
That looks correct to me